### PR TITLE
refactor(frontend): MaritalStatusService service methods naming convention

### PR DESCRIPTION
**Pull request by sebastien-comeau**
### Description

As per title to adhere to the [Service Method Naming Conventions](https://github.com/DTS-STN/canadian-dental-care-plan/tree/main/frontend#service-method-naming-conventions-find-vs-get-vs-list).

### Related Azure Boards Work Items

[AB#4577](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/4577)

### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/.server/domain/services/marital-status.service.ts [hunk 1]
```diff
@@ -26,32 +27,38 @@ export class MaritalStatusServiceImpl implements MaritalStatusService {
...
     // set moize options
-    this.findAll.options.maxAge = 1000 * this.serverConfig.LOOKUP_SVC_ALL_MARITAL_STATUSES_CACHE_TTL_SECONDS;
-    this.findById.options.maxAge = 1000 * this.serverConfig.LOOKUP_SVC_MARITAL_STATUS_CACHE_TTL_SECONDS;
+    this.listMaritalStatuses.options.maxAge = 1000 * this.serverConfig.LOOKUP_SVC_ALL_MARITAL_STATUSES_CACHE_TTL_SECONDS;
+    this.getMaritalStatusById.options.maxAge = 1000 * this.serverConfig.LOOKUP_SVC_MARITAL_STATUS_CACHE_TTL_SECONDS;
   }
 
-  private findAllImpl(): MaritalStatusDto[] {
+  listMaritalStatuses = moize(this.listMaritalStatusesImpl, {
+    onCacheAdd: () => this.log.info('Creating new findAll memo'),
```
Comment (nicholas-ly) [inline frontend/app/.server/domain/services/marital-status.service.ts:35]:
```suggestion
    onCacheAdd: () => this.log.info('Creating new listMaritalStatuses memo'),
```

File: frontend/app/.server/domain/services/marital-status.service.ts [hunk 2]
```diff
@@ -26,32 +27,38 @@ export class MaritalStatusServiceImpl implements MaritalStatusService {
...
+  });
+
+  getMaritalStatusById = moize(this.getMaritalStatusByIdImpl, {
+    maxSize: Infinity,
+    onCacheAdd: () => this.log.info('Creating new findById memo'),
```
Comment (nicholas-ly) [inline frontend/app/.server/domain/services/marital-status.service.ts:40]:
```suggestion
    onCacheAdd: () => this.log.info('Creating new getMaritalStatusById memo'),
```

**Review by nicholas-ly — APPROVED** [review]

File: frontend/app/.server/domain/services/marital-status.service.ts (hunk 1 above)
Comment (sebastien-comeau) [reply frontend/app/.server/domain/services/marital-status.service.ts]: Fixed

File: frontend/app/.server/domain/services/marital-status.service.ts (hunk 2 above)
Comment (sebastien-comeau) [reply frontend/app/.server/domain/services/marital-status.service.ts]: Fixed

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**